### PR TITLE
Deprecate cudf.isclose

### LINKED ISSUE #13593: Deprecate and remove `cudf.isclose`.
Currently we have a function `cudf.isclose` that is undocumented. It mirrors `cupy.isclose` but also has null support. Pandas does not have a similar `isclose` function. After discussion with @shwina, @galipremsagar, and @vyasr we decided to deprecate and remove this function.

See also: https://github.com/rapidsai/cudf/issues/5105

As part of the deprecation, we decided to add a warning message indicating how to support nulls while using `cupy.isclose`.

### PR DESCRIPTION
**Pull request by enoch-sit**
## Description
This PR adds deprecation warning to `cudf.isclose` and direct user to an alternative solution.
closes #13593

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: .devcontainer/cuda11.8-conda/devcontainer.json [hunk 1]
```diff
@@ -5,17 +5,17 @@
     "args": {
       "CUDA": "11.8",
       "PYTHON_PACKAGE_MANAGER": "conda",
-      "BASE": "rapidsai/devcontainers:24.12-cpp-cuda11.8-mambaforge-ubuntu22.04"
+      "BASE": "rapidsai/devcontainers:25.02-cpp-cuda11.8-mambaforge-ubuntu22.04"
```
Comment (KyleFromNVIDIA) [inline .devcontainer/cuda11.8-conda/devcontainer.json:8]: This PR targets 24.12. Why have all the dependencies been changed to 25.02?

Comment (bdice) [reply .devcontainer/cuda11.8-conda/devcontainer.json:8]: This appears to be an attempt at https://github.com/rapidsai/cudf/issues/13593, an issue I filed a while back. @the-gates-of-Zion I can help work with you on this, if you are interested. This should target `branch-25.02`, so I will retarget the PR.

File: python/cudf/cudf/core/series.py [hunk 2]
```diff
@@ -5322,6 +5322,29 @@ def isclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
     5    False
     dtype: bool
     """
+    warnings.warn(
+        "`cudf.close` is deprecated and will be removed in a future version of cudf. "
```
Comment (bdice) [inline python/cudf/cudf/core/series.py:5326]:
```suggestion
        "`cudf.isclose` is deprecated and will be removed in a future version of cudf. "
```

File: python/cudf/cudf/core/series.py [hunk 3]
```diff
@@ -5322,6 +5322,29 @@ def isclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
...
+
+        a_col = as_column(a)
+        a_array = cupy.asarray(a_col.data_array_view(mode="read"))
+
+        b_col = as_column(b)
+        b_array = cupy.asarray(b_col.data_array_view(mode="read"))
+
+        result = cp.isclose(a, b, equal_nan=True)
+        print(result)  # Output: [ True False  True]
+        ''',
```
Comment (bdice) [inline python/cudf/cudf/core/series.py:5345]: We noted in https://github.com/rapidsai/cudf/issues/13593: "As part of the deprecation, we decided to add a warning message indicating how to support nulls while using cupy.isclose." However, I am not sure if this is the right approach.

The code we want to give to the user as a replacement for this deprecated code path is essentially the current implementation in this method. Basically it has two steps:
- Call `cupy.isclose` to see if the values are "close" to each other. This does not account for null values in the input.
- Compare the null masks of each input. If `equal_nan` is `False`, we mark _all_ null values as `False` ("not close") in the result. If `equal_nan` is `True`, we set the values where one column's input is null as `False` and the values where both columns' input are null as `True`.

I would suggest we add this snippet to the page "Working with missing data": https://docs.rapids.ai/api/cudf/stable/user_guide/missing-data/

Then we can link to this snippet from the deprecated method's docstring and in the deprecation warning.

Does that make sense? Let me know if you have questions.

**Review by bdice — CHANGES_REQUESTED** [review]

**Comment by isVoid** [comment]
Closing the PR due to I accidentally rewrote the history of the previous commits.

**Comment by vyasr** [comment]
@isVoid do you want to open a new PR here?

**Pull request closed without merging**